specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-04-01-preview/registries.json

```diff
@@ -507,56 +567,141 @@
+          "type": "string",
+          "x-nullable": true
+        },
+        "location": {
+          "description": "Same as workspace location.",
+          "type": "string",
+          "x-nullable": true
+        },
+        "properties": {
+          "description": "Properties of the Private Endpoint Connection",
+          "$ref": "#/definitions/PrivateEndpointConnectionProperties",
+          "x-nullable": true
         }
-      ],
+      },
+      "additionalProperties": false
+    },
+    "PrivateEndpointConnectionProperties": {
+      "type": "object",
       "properties": {
-        "publicNetworkAccess": {
+        "groupIds": {
+          "description": "The group ids",
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "x-nullable": true
+        },
+        "privateEndpoint": {
+          "description": "The PE network resource that is linked to this PE connection.",
+          "$ref": "#/definitions/PrivateEndpointResource",
+          "x-nullable": true
+        },
+        "privateLinkServiceConnectionState": {
+          "description": "The connection state.",
+          "$ref": "#/definitions/PrivateLinkServiceConnectionState",
+          "x-nullable": true
+        },
+        "provisioningState": {
+          "description": "One of null, \"Succeeded\", \"Provisioning\", \"Failed\". While not approved, it's null.",
+          "type": "string",
+          "x-nullable": true
+        }
+      },
+      "additionalProperties": false
+    },
+    "PrivateEndpointResource": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "e.g. /subscriptions/{networkSubscriptionId}/resourceGroups/{rgName}/providers/Microsoft.Network/privateEndpoints/{privateEndpointName}",
           "type": "string",
           "x-nullable": true
         },
+        "subnetArmId": {
+          "description": "The subnetId that the private endpoint is connected to.",
+          "type": "string",
+          "x-nullable": true
+        }
+      },
+      "additionalProperties": false
+    },
+    "PrivateLinkServiceConnectionState": {
+      "type": "object",
+      "properties": {
+        "actionsRequired": {
+          "description": "Some RP chose \"None\". Other RPs use this for region expansion.",
+          "type": "string",
+          "x-nullable": true
+        },
+        "description": {
+          "description": "User-defined message that, per NRP doc, may be used for approval-related message.",
+          "type": "string",
+          "x-nullable": true
+        },
+        "status": {
+          "description": "Connection status of the service consumer with the service provider",
+          "$ref": "#/definitions/EndpointServiceConnectionStatus"
+        }
+      },
+      "additionalProperties": false
+    },
```

Comment:
You can use the definition from [https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v3/privatelinks.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v3/privatelinks.json)

Reply:
actually, taking a look at it now I see that in PrivateLinkServiceConnectionState, that status is property is defined by PrivateEndpointServiceConnectionStatus which includes only 3 states (pending, approved, rejected) but the way it is defined in Common has a 4th state: disconnected, [EndpointServiceConnectionStatus.cs](https://msdata.visualstudio.com/Vienna/_git/vienna?path=/src/azureml-api/src/Common/PrivateLink.Contracts/Nrp/EndpointServiceConnectionStatus.cs&_a=contents&version=GBmaster)